docs/architecture/stack/trace-expansion-proving/prover-limits.mdx

```diff
@@ -0,0 +1,36 @@
+---
+title: Module limits
+description: How module limits ensure the prover can generate proofs.
+sidebar_position: 3
+---
+
+Linea has a built-in mechanism for ensuring that the prover can continue to generate proofs
+without increasing proof complexity to such a level that current infrastructure cannot process them. 
+This mechanism involves module limits. 
+
+Modules are the sections into which Linea's arithmetization specification is divided. Each module 
+represents a collection of operations (opcodes) at the EVM level, or corresponds to a specific 
+function of Linea's zkEVM. Some of the modules process transaction data; some manage zkEVM memory; 
+others coordinate interactions between modules.
+
+Regardless of its purpose, each module has a specific limit to the amount of lines of data that 
+its operations can generate. You can view each module's limits in the [Linea source code](https://github.com/Consensys/linea-monorepo/blob/main/config/common/traces-limits-besu-v2.toml).
+Every time one of the operations in a module is used, it generates additional lines.
```

Comment:
It _may_ generate additional lines, this depends on the module. For instance for modules that deal with 'simple' instructions à la `ADD`, `SUB`, `MUL`, `EXP`, `AND`, `OR`, `XOR`, `NOT`, `BYTE`, `SIGNEXTEND`, `SHL`, `SHR`, `SAR`, `MOD`, `DIV`, `SMOD`, `SDIV`, `ADDMOD`, `MULMOD`, ... among others only get traced once so repeat calls with the same arguments don't get traced multiple times.